relay_messages.go

```diff
@@ -134,22 +141,38 @@ func newLazyCallReq(f *Frame) (*lazyCallReq, error) {
 	}
 
 	// csumtype:1 (csum:4){0,1} arg1~2 arg2~2 arg3~2
-	checkSumType := ChecksumType(rbuf.ReadSingleByte())
-	rbuf.ReadBytes(checkSumType.ChecksumSize())
+	cr.checksumTypeOffset = rbufOffset()
+	cr.checksumType = ChecksumType(rbuf.ReadSingleByte())
+	rbuf.ReadBytes(cr.checksumType.ChecksumSize())
 
 	// arg1~2
 	arg1Len := int(rbuf.ReadUint16())
 	cr.method = rbuf.ReadBytes(arg1Len)
 
 	// arg2~2
-	arg2Len := int(rbuf.ReadUint16())
-	cr.arg2StartOffset = int(headerStartIndex) + rbuf.BytesRead()
-	// TODO(echung): deprecate the arg2 offsets and store just the arg2 slice instead
-	// Read past arg2
-	rbuf.ReadBytes(arg2Len)
+	arg2Len := rbuf.ReadUint16()
+	cr.arg2StartOffset = rbufOffset()
 	cr.arg2EndOffset = cr.arg2StartOffset + arg2Len
+
 	// arg2 is fragmented if we don't see arg3 in this frame.
-	cr.isArg2Fragmented = rbuf.BytesRemaining() == 0 && cr.HasMoreFragments()
+	if uint16(rbuf.BytesRemaining()) <= arg2Len {
```

Comment:
if the frame says it contains X bytes for arg2, then it must contain all X bytes, otherwise it's a protocol error. So effectively: `BytesRemaining` is always >= `arg2Len`. Can we simplify this to:
```go
rbuf.Skip(arg2Len)
cr.isArg2Fragmented = rbuf.BytesRemaining() == 0 && cr.HasMoreFragments()
```